action.yml

```diff
@@ -34,6 +34,19 @@ inputs:
     required: false
     default: "1"
 
+  min-versions-to-keep:
+    description: >
+      Number of versions to keep starting with the latest version
+    required: false
+    default: "1"
+
+  ignore-versions:
+    description: >
+      Comma seperated regex patterns of versions to ignore.
+      Defaults to all major and minor versions and only deletes pre-release and patch versions.
+    required: false
+    default: "^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0+)$"
```

Comment:
NuGet is pretty loose about what version strings are valid. See here for more info:
[I think NuGet users ](https://docs.microsoft.com/en-us/nuget/concepts/package-versioning#where-nugetversion-diverges-from-semantic-versioning)

In particular:

> `NuGetVersion` only requires the major segment to be defined. All others are optional, and are equivalent to zero. This means that `1`, `1.0`, `1.0.0`, and `1.0.0.0` are all accepted and equal.

I'd be include to default to keeping all versions that only include number segments.

That way we wouldn't delete `1`, `1.0`, `1.0.0`, and `1.0.0.0`, but we would delete `1.0.0-foo` and `1.0.0.pre.foo` (Ruby style).

Does that make sense?

Reply:
This does makes sense as we are looking to delete only pre-release versions. Updating the regex to include versions like `1` `1.0` `1.0.0` `1.0.0.0` and so on. Ignoring rest of the package versions like `1.0.0-foo` and others.